### DISCUSSION TITLE
Squash & Merge (as a Pull Request) missing commit from tip of branch

### DISCUSSION THREAD

**Discussion by carlleeswanson** [community/community#22840] [New to GitHub]
<p>Yesterday, in response to a Pull Request, I did a Squash/Merge from a dev_branch to master.</p>
<p>When I look at the Pull Request (now closed), it shows the “Merged” tag and correctly lists the 10 commits (shows commit messages/ids) that went into the merge. As expected, the last one listed is the tip of dev_branch. When I select the link to that commit, it correctly shows that the last commit on dev_branch consisted exclusively of the deletion of 3 unused files.</p>
<p>When I go to the master branch, and examine the commit corresponding to the aforementioned merge/pull request, the comments show the appropriate/same 10 commit messages from the merge. But the list of files changed <em>does not</em> include the 3 that were deleted by the last commit on dev_branch. So, the files are still present on master.</p>
<p>There seems to be a mis-match between what happened with the Squash/Merge and what happened with the Pull.</p>
<p>Is this a bug in GitHub’s Pull/Merge functionality? Or am I missing something?</p>

**Comment by carlleeswanson** [comment, marked as answer]
<p>Okay, I can answer this myself…<br>
TLDR: noob error.</p>
<p>Full answer:<br>
I had copied an intermediate commit on the dev_branch to a dev_branch2.<br>
dev_branch2 at that point included the <em>creation</em> of the aforementioned 3 files.<br>
dev_branch2 was subsequently merged to master. So far so good…</p>
<p>The problem I described arose from me squash/merging the tip of dev_branch, which included the entire set of commits, including:</p>
<ul>
<li>one commit (also on dev_branch2) that <em>created</em> the 3 files</li>
<li>another commit that <em>deleted</em> the 3 files</li>
</ul>
<p>The squash/merge correctly saw that the files were both created and deleted, which effectively squashed to no change. That’s why the final commit didn’t include the delete.</p>
<p>Ultimately, the problem occurred because of the way I did the two squash/merges. In the future I will use a different process, namely cherry-picking the later dev_branch commits to a new branch, then squash/merging that.</p>
